swarm/network/stream/syncer.go

```diff
@@ -180,17 +191,27 @@ func RegisterSwarmSyncerClient(streamer *Registry, netStore *storage.NetStore) {
 	})
 }
 
-// NeedData
-func (s *SwarmSyncerClient) NeedData(ctx context.Context, key []byte) (wait func(context.Context) error) {
-	return s.netStore.FetchFunc(ctx, key)
+func (s *SwarmSyncerClient) NeedData(ctx context.Context, key []byte) (loaded bool, wait func(context.Context) error) {
```

Comment:
this function needs a better name, IMHO

Reply:
@gluk256 this would be too many changes for this PR. `NeedData` is part of the interface of the syncer, that has been in the codebase for a while.